File: LeetCode/remove_element.py
#!/usr/bin/env python
#coding: utf-8

class Solution:
    # @param    A       a list of integers
    # @param    elem    an integer, value need to be removed
    # @return an integer
    def removeElement(self, A, elem):
        i, la = 0, len(A)
        while i < la:
            while i < la and A[i] != elem:
                i += 1
            if i == la: break

            j = i
            while j < la and A[j] == elem:
                j += 1
            
            rm_n = j - i
            for k in range(j, la):
                A[k - rm_n] = A[k]

            la -= rm_n
        return la


if __name__ == '__main__':
    A = [1, 2, 3, 3, 4, 3, 5, 3, 3]
    s = Solution()
    print(s.removeElement(A, 3))
    print(A)

File: LeetCode/Java/Solution/src/ios/ac/cn/RemoveElement.java
package ios.ac.cn;

public class RemoveElement {
	public int removeElement(int[] A, int elem) {
		if (A == null) {
			return 0;
		}

		int curLen = 0, curP = 0;
		while (curP < A.length) {
			if (A[curP] != elem) {
				A[curLen++] = A[curP];
			}
			curP++;
		}
		return curLen;
	}
}
